salt/metalk8s/kubernetes/coredns/deployed.sls

```diff
@@ -6,111 +6,129 @@
 {% set context = "kubernetes-admin@kubernetes" %}
 
 Create coredns ConfigMap:
-  metalk8s_kubernetes.configmap_present:
-    - name: coredns
-    - namespace: kube-system
+  metalk8s_kubernetes.object_present:
+    - manifest:
+        apiVersion: v1
+        kind: ConfigMap
+        metadata:
+          name: coredns
+          namespace: kube-system
+        data:
+          Corefile: |
+            .:53 {
+                errors
+                health
+                kubernetes {{ coredns.cluster_domain }} {{ coredns.reverse_cidrs }} {
+                  pods insecure
+                  upstream
+                  fallthrough in-addr.arpa ip6.arpa
+                }
+                prometheus :9153
+                proxy . /etc/resolv.conf
+                cache 30
+                loop
+                reload
+                loadbalance
+            }
     - kubeconfig: {{ kubeconfig }}
     - context: {{ context }}
-    - data:
-        Corefile: |
-          .:53 {
-              errors
-              health
-              kubernetes {{ coredns.cluster_domain }} {{ coredns.reverse_cidrs }} {
-                pods insecure
-                upstream
-                fallthrough in-addr.arpa ip6.arpa
-              }
-              prometheus :9153
-              proxy . /etc/resolv.conf
-              cache 30
-              loop
-              reload
-              loadbalance
-          }
 
 Create coredns deployment:
-  metalk8s_kubernetes.deployment_present:
-    - name: coredns
-    - namespace: kube-system
+  metalk8s_kubernetes.object_present:
+    - name: salt://{{ slspath }}/files/coredns-deployment.yaml.j2
```

Comment:
Why call this argument `name` and not `source`, as was done before? I feel like it is less explicit this way.

Reply:
`name` for state is better so here we could use simple syntax to deploy object
```
/tmp/toto-manifest.yaml:
  metalk8s_kubernetes.object_present
```

Reply:
Beurk.

Reply:
It's something salt suport btw but it's maybe more usefull when you do a `state.single`
With `name` as source:
```
salt-run salt.cmd state.single metalk8s_kubernetes.object_present /tmp/toto-manifest.yaml
```
With dedicated `source` argument:
```
salt-run salt.cmd state.single metalk8s_kubernetes.object_present '<whatever_you_want_as_name_is _mandatory>' source='/tmp/toto-manifest.yaml'
```

Reply:
I'm with @TeddyAndrieux on this one. It may not be 'great' but it's how things are done all over Salt, so 'name' being the 'main argument', in things that take some filename this being said filename, is 'standard expected behaviour' of a state.

Reply:
Mmmh OK, let's keep doing things as Salt does. I just really dislike how `name` can be the actual object name or the path to its manifest.